### ISSUE TITLE
Meeting 2021-11-18

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#373]
- progress on deactorizer?
  - how far along are we? 10? 20%? 30%?
  - what conceptual questions are there?
-

**Comment by plajjan** [comment]
Big conceptual issue for the deactorizer that we talked about. An actor method is of type "action". Regardless if it is a function on a local actor, which runs sync, or if it on a remote actor, which runs async in certain cases and sync in other, the type system has just seen an action. Handling sync/async has been left to the deactorizer. Turns out we get some weird cases, like passing a function as an argument to a local function A will have the deactorizer wrap it accordingly, if that local function A then passes the function on to another remote actor Bopa as an argument, the deactorizer will wrap it again, which means the remote actor Bopa will send messages through our local actor even if they are actually going to a third actor. We talked about the semantics. Decided upon what we want. Not possible to hide this stuff from our users / developers. Must have it front and center.
